xline-client/src/lib.rs

```diff
@@ -403,8 +425,19 @@ impl ClientOptions {
     #[must_use]
     pub fn with_client_config(self, client_config: ClientConfig) -> Self {
         Self {
-            user: self.user,
             client_config,
+            ..self
+        }
+    }
+
+    /// Set `tls_config`
+    #[inline]
+    #[must_use]
+    #[cfg(not(madsim))]
+    pub fn with_tls_config(self, tls_config: ClientTlsConfig) -> Self {
+        Self {
+            tls_config: Some(tls_config),
+            ..self
```

Comment:
Should `ClientTlsConfig` be a part of `ClientConfig`?

Reply:
Currently, the content in `ClientConfig` is directly obtained from command line arguments or configuration files, while `ClientTlsConfig` needs to read the certificate file and cannot be obtained directly, they can't contain each other for now, but our config logic may be modified when config is refactored in the future